rclrs/src/parameter/value.rs

```diff
@@ -26,27 +28,257 @@ pub enum ParameterValue {
+            _ => None,
+        }
+    }
+
+    fn kind() -> ParameterKind {
+        ParameterKind::Bool
+    }
+}
+
+impl ParameterVariant for i64 {
+    type Range = ParameterRange<i64>;
+    fn maybe_from(value: ParameterValue) -> Option<Self> {
+        match value {
+            ParameterValue::Integer(v) => Some(v),
+            _ => None,
+        }
+    }
+
+    fn kind() -> ParameterKind {
+        ParameterKind::Integer
+    }
+}
+
+impl ParameterVariant for f64 {
+    type Range = ParameterRange<f64>;
+    fn maybe_from(value: ParameterValue) -> Option<Self> {
+        match value {
+            ParameterValue::Double(v) => Some(v),
+            _ => None,
+        }
+    }
+
+    fn kind() -> ParameterKind {
+        ParameterKind::Double
+    }
+}
+
+impl ParameterVariant for Arc<str> {
+    type Range = ();
+    fn maybe_from(value: ParameterValue) -> Option<Self> {
+        match value {
+            ParameterValue::String(v) => Some(v),
+            _ => None,
+        }
+    }
+
+    fn kind() -> ParameterKind {
+        ParameterKind::String
+    }
+}
+
+impl ParameterVariant for Arc<[u8]> {
+    type Range = ();
+    fn maybe_from(value: ParameterValue) -> Option<Self> {
+        match value {
+            ParameterValue::ByteArray(v) => Some(v),
+            _ => None,
+        }
+    }
+
+    fn kind() -> ParameterKind {
+        ParameterKind::ByteArray
+    }
+}
+
+impl ParameterVariant for Arc<[bool]> {
+    type Range = ();
+    fn maybe_from(value: ParameterValue) -> Option<Self> {
+        match value {
+            ParameterValue::BoolArray(v) => Some(v),
+            _ => None,
+        }
+    }
+
+    fn kind() -> ParameterKind {
+        ParameterKind::BoolArray
+    }
+}
+
+impl ParameterVariant for Arc<[i64]> {
+    type Range = ();
+    fn maybe_from(value: ParameterValue) -> Option<Self> {
+        match value {
+            ParameterValue::IntegerArray(v) => Some(v),
+            _ => None,
+        }
+    }
+
+    fn kind() -> ParameterKind {
+        ParameterKind::IntegerArray
+    }
+}
+
+impl ParameterVariant for Arc<[f64]> {
+    type Range = ();
+    fn maybe_from(value: ParameterValue) -> Option<Self> {
+        match value {
+            ParameterValue::DoubleArray(v) => Some(v),
+            _ => None,
+        }
+    }
+
+    fn kind() -> ParameterKind {
+        ParameterKind::DoubleArray
+    }
+}
+
+impl ParameterVariant for Arc<[Arc<str>]> {
+    type Range = ();
+    fn maybe_from(value: ParameterValue) -> Option<Self> {
+        match value {
+            ParameterValue::StringArray(v) => Some(v),
+            _ => None,
+        }
+    }
+
+    fn kind() -> ParameterKind {
+        ParameterKind::StringArray
+    }
+}
+
+impl ParameterVariant for ParameterValue {
+    type Range = ParameterRanges;
+    fn maybe_from(value: ParameterValue) -> Option<Self> {
+        Some(value)
+    }
+
+    fn kind() -> ParameterKind {
+        ParameterKind::Dynamic
+    }
```

Comment:
Are there good use cases for dynamic parameters? This functionality always seemed like an antipattern to me, so I left it out from my prototype to keep complexity smaller, at least initially.

Reply:
I... Honestly don't know of any real use case but I thought I would add the functionality since it is present in the other libraries.
It is also part of [ParameterDescriptor](https://github.com/ros2/rcl_interfaces/blob/fbd32f8b8ef90fb9cefdc90d8d2f5b992fd843bb/rcl_interfaces/msg/ParameterDescriptor.msg#L25) and I thought we should have an implementation for all the fields in the message so when we implement services and users call `/describe_parameters` we have an implementation for all the features in the message fields.
The nice aspect of the proposed implementation is that it is just treated as a different enum value and not as a boolean flag that we need to check all over the place so it doesn't really clutter the code or introduce potential errors too much. If there is consensus to remove it I'm happy to do so though